occurrence-download/src/main/java/org/gbif/occurrence/download/file/specieslist/SpeciesListDownloadAggregator.java

```diff
@@ -0,0 +1,148 @@
+    this.configuration = configuration;
+    this.workflowConfiguration = workflowConfiguration;
+    outputFileName =
+      configuration.getDownloadTempDir() + Path.SEPARATOR + configuration.getDownloadKey() + CSV_EXTENSION;
+    this.occurrenceDownloadService = occurrenceDownloadService;
+  }
+
+  
+  @Override
+  public void aggregate(List<Result> results) {
+    try {
+      if (!results.isEmpty()) {
+        mergeResults(results);
+        
+      }
+      
+      SimpleCsvArchiveBuilder.withHeader(DownloadTerms.SPECIES_LIST_DOWNLOAD_TERMS).mergeToZip(FileSystem.getLocal(new Configuration()).getRawFileSystem(),
+                                         DownloadFileUtils.getHdfs(workflowConfiguration.getHdfsNameNode()),
+                                         configuration.getDownloadTempDir(),
+                                         workflowConfiguration.getHdfsOutputPath(),
+                                         configuration.getDownloadKey(),
+                                         ModalZipOutputStream.MODE.DEFAULT);
+      //Delete the temp directory
+      FileUtils.deleteDirectoryRecursively(Paths.get(configuration.getDownloadTempDir()).toFile());
+    } catch (IOException ex) {
+      LOG.error("Error aggregating download files", ex);
+      throw Throwables.propagate(ex);
+    }
+    
+  }
+  
+  /**
+   * Merges the files of each job into a single CSV file.
+   */
+  private void mergeResults(List<Result> results) {
+    // Results are sorted to respect the original ordering
+    Collections.sort(results);
+    DatasetUsagesCollector datasetUsagesCollector = new DatasetUsagesCollector();
+    List<Map<String, String>> aggregateSpeciesList = new ArrayList<>();
+    for (Result result : results) {
+      datasetUsagesCollector.sumUsages(result.getDatasetUsages());
+      datasetUsagesCollector.mergeLicenses(result.getDatasetLicenses());
+      aggregateSpeciesList.addAll(result.getSpeciesListCollector().getDistinctSpecies());
+    }
+
+    try (ICsvMapWriter csvMapWriter =
+        new CsvMapWriter(new FileWriterWithEncoding(outputFileName, StandardCharsets.UTF_8),
+            CsvPreference.TAB_PREFERENCE)) {
+      List<Map<String, String>> distinctSpecies = new SpeciesListCollector().computeDistinctSpecies(aggregateSpeciesList).getDistinctSpecies();
+      distinctSpecies.iterator().forEachRemaining(speciesInfo -> {
+        try {
+          csvMapWriter.write(speciesInfo, COLUMNS);
+        } catch (IOException e) {
+          LOG.error("Error merging results", e);
+          throw Throwables.propagate(e);
+        }
+      });
+      occurrenceDownloadService.createUsages(configuration.getDownloadKey(),
+          datasetUsagesCollector.getDatasetUsages());
+      persistDownloadLicense(configuration.getDownloadKey(),
+          datasetUsagesCollector.getDatasetLicenses());
+      Properties properties = PropertiesUtil.loadProperties(DownloadWorkflowModule.CONF_FILE);
```

Comment:
to avoid all this we can make SpeciesCount to be SpeciesCountUpdate (or something similar), and pass the occurrenceDownloadService as param to perform the update there, in this case we avoid to instantiate that service twice and do all that registry config look-up